{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(11408, 11)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>uTAR</th>\n",
       "      <th>fasta</th>\n",
       "      <th>fasta peak</th>\n",
       "      <th>peak blast result</th>\n",
       "      <th>cell type</th>\n",
       "      <th>p value</th>\n",
       "      <th>avg. logFC</th>\n",
       "      <th>% expression in cell type</th>\n",
       "      <th>% expression in other cells</th>\n",
       "      <th>p value adj.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Antoine_Bladder_10X</td>\n",
       "      <td>NC-033671.1-80333049-80333549---18-0</td>\n",
       "      <td>NC_033671.1:80333049-80333549</td>\n",
       "      <td>NC_033671.1:80333248-80333394</td>\n",
       "      <td>Apteryx australis mantelli genome assembly Apt...</td>\n",
       "      <td>B cell</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.801622</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Antoine_Bladder_10X</td>\n",
       "      <td>NC-033666.1-51567149-51568449-+-84-0</td>\n",
       "      <td>NC_033666.1:51567149-51568449</td>\n",
       "      <td>NC_033666.1:51568050-51568213</td>\n",
       "      <td>undetermined</td>\n",
       "      <td>B cell</td>\n",
       "      <td>9.602478e-60</td>\n",
       "      <td>0.534839</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.002</td>\n",
       "      <td>2.880340e-54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Antoine_Bladder_10X</td>\n",
       "      <td>NC-033672.1-36774049-36774849---67-0</td>\n",
       "      <td>NC_033672.1:36774049-36774849</td>\n",
       "      <td>NC_033672.1:36774301-36774386</td>\n",
       "      <td>Microcebus murinus chromosome chr16 clone CH25...</td>\n",
       "      <td>B cell</td>\n",
       "      <td>1.143571e-14</td>\n",
       "      <td>0.535301</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.013</td>\n",
       "      <td>3.430234e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Antoine_Bladder_10X</td>\n",
       "      <td>NC-033676.1-58848749-58864449---1010-0</td>\n",
       "      <td>NC_033676.1:58848749-58864449</td>\n",
       "      <td>NC_033676.1:58858363-58860157</td>\n",
       "      <td>PREDICTED: Propithecus coquereli sialophorin (...</td>\n",
       "      <td>B cell</td>\n",
       "      <td>4.116833e-09</td>\n",
       "      <td>0.865333</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.021</td>\n",
       "      <td>1.234877e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Antoine_Bladder_10X</td>\n",
       "      <td>NC-033666.1-87778599-87781699---78-0</td>\n",
       "      <td>NC_033666.1:87778599-87781699</td>\n",
       "      <td>NC_033666.1:87780600-87780897</td>\n",
       "      <td>Microcebus murinus DNA, homologous to human DS...</td>\n",
       "      <td>B cell</td>\n",
       "      <td>5.894225e-07</td>\n",
       "      <td>0.576084</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.029</td>\n",
       "      <td>1.768020e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample                                    uTAR  \\\n",
       "0  Antoine_Bladder_10X    NC-033671.1-80333049-80333549---18-0   \n",
       "1  Antoine_Bladder_10X    NC-033666.1-51567149-51568449-+-84-0   \n",
       "2  Antoine_Bladder_10X    NC-033672.1-36774049-36774849---67-0   \n",
       "3  Antoine_Bladder_10X  NC-033676.1-58848749-58864449---1010-0   \n",
       "4  Antoine_Bladder_10X    NC-033666.1-87778599-87781699---78-0   \n",
       "\n",
       "                           fasta                     fasta peak  \\\n",
       "0  NC_033671.1:80333049-80333549  NC_033671.1:80333248-80333394   \n",
       "1  NC_033666.1:51567149-51568449  NC_033666.1:51568050-51568213   \n",
       "2  NC_033672.1:36774049-36774849  NC_033672.1:36774301-36774386   \n",
       "3  NC_033676.1:58848749-58864449  NC_033676.1:58858363-58860157   \n",
       "4  NC_033666.1:87778599-87781699  NC_033666.1:87780600-87780897   \n",
       "\n",
       "                                   peak blast result cell type       p value  \\\n",
       "0  Apteryx australis mantelli genome assembly Apt...    B cell  0.000000e+00   \n",
       "1                                       undetermined    B cell  9.602478e-60   \n",
       "2  Microcebus murinus chromosome chr16 clone CH25...    B cell  1.143571e-14   \n",
       "3  PREDICTED: Propithecus coquereli sialophorin (...    B cell  4.116833e-09   \n",
       "4  Microcebus murinus DNA, homologous to human DS...    B cell  5.894225e-07   \n",
       "\n",
       "   avg. logFC  % expression in cell type  % expression in other cells  \\\n",
       "0    0.801622                        0.4                        0.000   \n",
       "1    0.534839                        0.4                        0.002   \n",
       "2    0.535301                        0.4                        0.013   \n",
       "3    0.865333                        0.4                        0.021   \n",
       "4    0.576084                        0.4                        0.029   \n",
       "\n",
       "   p value adj.  \n",
       "0  0.000000e+00  \n",
       "1  2.880340e-54  \n",
       "2  3.430234e-09  \n",
       "3  1.234877e-03  \n",
       "4  1.768020e-01  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "utar_df = pd.read_csv('/home/olga/googledrive/.shared/MLCAuTARs.csv')\n",
    "print(utar_df.shape)\n",
    "utar_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I have an updated set of uTARs generated from Camille's updated cell annotations. Can you re-run your pipeline with this new set of uTARs? There are 3935 unique uTARs in the new data. Specifically, can you run your pipeline with the \"fasta peak\" fasta references?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>uTAR</th>\n",
       "      <th>fasta</th>\n",
       "      <th>fasta peak</th>\n",
       "      <th>peak blast result</th>\n",
       "      <th>cell type</th>\n",
       "      <th>p value</th>\n",
       "      <th>avg. logFC</th>\n",
       "      <th>% expression in cell type</th>\n",
       "      <th>% expression in other cells</th>\n",
       "      <th>p value adj.</th>\n",
       "      <th>chrom</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Antoine_Bladder_10X</td>\n",
       "      <td>NC-033671.1-80333049-80333549---18-0</td>\n",
       "      <td>NC_033671.1:80333049-80333549</td>\n",
       "      <td>NC_033671.1:80333248-80333394</td>\n",
       "      <td>Apteryx australis mantelli genome assembly Apt...</td>\n",
       "      <td>B cell</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.801622</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NC_033671.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Antoine_Bladder_10X</td>\n",
       "      <td>NC-033666.1-51567149-51568449-+-84-0</td>\n",
       "      <td>NC_033666.1:51567149-51568449</td>\n",
       "      <td>NC_033666.1:51568050-51568213</td>\n",
       "      <td>undetermined</td>\n",
       "      <td>B cell</td>\n",
       "      <td>9.602478e-60</td>\n",
       "      <td>0.534839</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.002</td>\n",
       "      <td>2.880340e-54</td>\n",
       "      <td>NC_033666.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Antoine_Bladder_10X</td>\n",
       "      <td>NC-033672.1-36774049-36774849---67-0</td>\n",
       "      <td>NC_033672.1:36774049-36774849</td>\n",
       "      <td>NC_033672.1:36774301-36774386</td>\n",
       "      <td>Microcebus murinus chromosome chr16 clone CH25...</td>\n",
       "      <td>B cell</td>\n",
       "      <td>1.143571e-14</td>\n",
       "      <td>0.535301</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.013</td>\n",
       "      <td>3.430234e-09</td>\n",
       "      <td>NC_033672.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Antoine_Bladder_10X</td>\n",
       "      <td>NC-033676.1-58848749-58864449---1010-0</td>\n",
       "      <td>NC_033676.1:58848749-58864449</td>\n",
       "      <td>NC_033676.1:58858363-58860157</td>\n",
       "      <td>PREDICTED: Propithecus coquereli sialophorin (...</td>\n",
       "      <td>B cell</td>\n",
       "      <td>4.116833e-09</td>\n",
       "      <td>0.865333</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.021</td>\n",
       "      <td>1.234877e-03</td>\n",
       "      <td>NC_033676.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Antoine_Bladder_10X</td>\n",
       "      <td>NC-033666.1-87778599-87781699---78-0</td>\n",
       "      <td>NC_033666.1:87778599-87781699</td>\n",
       "      <td>NC_033666.1:87780600-87780897</td>\n",
       "      <td>Microcebus murinus DNA, homologous to human DS...</td>\n",
       "      <td>B cell</td>\n",
       "      <td>5.894225e-07</td>\n",
       "      <td>0.576084</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.029</td>\n",
       "      <td>1.768020e-01</td>\n",
       "      <td>NC_033666.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample                                    uTAR  \\\n",
       "0  Antoine_Bladder_10X    NC-033671.1-80333049-80333549---18-0   \n",
       "1  Antoine_Bladder_10X    NC-033666.1-51567149-51568449-+-84-0   \n",
       "2  Antoine_Bladder_10X    NC-033672.1-36774049-36774849---67-0   \n",
       "3  Antoine_Bladder_10X  NC-033676.1-58848749-58864449---1010-0   \n",
       "4  Antoine_Bladder_10X    NC-033666.1-87778599-87781699---78-0   \n",
       "\n",
       "                           fasta                     fasta peak  \\\n",
       "0  NC_033671.1:80333049-80333549  NC_033671.1:80333248-80333394   \n",
       "1  NC_033666.1:51567149-51568449  NC_033666.1:51568050-51568213   \n",
       "2  NC_033672.1:36774049-36774849  NC_033672.1:36774301-36774386   \n",
       "3  NC_033676.1:58848749-58864449  NC_033676.1:58858363-58860157   \n",
       "4  NC_033666.1:87778599-87781699  NC_033666.1:87780600-87780897   \n",
       "\n",
       "                                   peak blast result cell type       p value  \\\n",
       "0  Apteryx australis mantelli genome assembly Apt...    B cell  0.000000e+00   \n",
       "1                                       undetermined    B cell  9.602478e-60   \n",
       "2  Microcebus murinus chromosome chr16 clone CH25...    B cell  1.143571e-14   \n",
       "3  PREDICTED: Propithecus coquereli sialophorin (...    B cell  4.116833e-09   \n",
       "4  Microcebus murinus DNA, homologous to human DS...    B cell  5.894225e-07   \n",
       "\n",
       "   avg. logFC  % expression in cell type  % expression in other cells  \\\n",
       "0    0.801622                        0.4                        0.000   \n",
       "1    0.534839                        0.4                        0.002   \n",
       "2    0.535301                        0.4                        0.013   \n",
       "3    0.865333                        0.4                        0.021   \n",
       "4    0.576084                        0.4                        0.029   \n",
       "\n",
       "   p value adj.        chrom  \n",
       "0  0.000000e+00  NC_033671.1  \n",
       "1  2.880340e-54  NC_033666.1  \n",
       "2  3.430234e-09  NC_033672.1  \n",
       "3  1.234877e-03  NC_033676.1  \n",
       "4  1.768020e-01  NC_033666.1  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "col_to_use = 'fasta peak'\n",
    "utar_df['chrom'] = utar_df[col_to_use].str.split(':').str[0]\n",
    "utar_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>uTAR</th>\n",
       "      <th>fasta</th>\n",
       "      <th>fasta peak</th>\n",
       "      <th>peak blast result</th>\n",
       "      <th>cell type</th>\n",
       "      <th>p value</th>\n",
       "      <th>avg. logFC</th>\n",
       "      <th>% expression in cell type</th>\n",
       "      <th>% expression in other cells</th>\n",
       "      <th>p value adj.</th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Antoine_Bladder_10X</td>\n",
       "      <td>NC-033671.1-80333049-80333549---18-0</td>\n",
       "      <td>NC_033671.1:80333049-80333549</td>\n",
       "      <td>NC_033671.1:80333248-80333394</td>\n",
       "      <td>Apteryx australis mantelli genome assembly Apt...</td>\n",
       "      <td>B cell</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.801622</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NC_033671.1</td>\n",
       "      <td>80333248</td>\n",
       "      <td>80333394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Antoine_Bladder_10X</td>\n",
       "      <td>NC-033666.1-51567149-51568449-+-84-0</td>\n",
       "      <td>NC_033666.1:51567149-51568449</td>\n",
       "      <td>NC_033666.1:51568050-51568213</td>\n",
       "      <td>undetermined</td>\n",
       "      <td>B cell</td>\n",
       "      <td>9.602478e-60</td>\n",
       "      <td>0.534839</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.002</td>\n",
       "      <td>2.880340e-54</td>\n",
       "      <td>NC_033666.1</td>\n",
       "      <td>51568050</td>\n",
       "      <td>51568213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Antoine_Bladder_10X</td>\n",
       "      <td>NC-033672.1-36774049-36774849---67-0</td>\n",
       "      <td>NC_033672.1:36774049-36774849</td>\n",
       "      <td>NC_033672.1:36774301-36774386</td>\n",
       "      <td>Microcebus murinus chromosome chr16 clone CH25...</td>\n",
       "      <td>B cell</td>\n",
       "      <td>1.143571e-14</td>\n",
       "      <td>0.535301</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.013</td>\n",
       "      <td>3.430234e-09</td>\n",
       "      <td>NC_033672.1</td>\n",
       "      <td>36774301</td>\n",
       "      <td>36774386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Antoine_Bladder_10X</td>\n",
       "      <td>NC-033676.1-58848749-58864449---1010-0</td>\n",
       "      <td>NC_033676.1:58848749-58864449</td>\n",
       "      <td>NC_033676.1:58858363-58860157</td>\n",
       "      <td>PREDICTED: Propithecus coquereli sialophorin (...</td>\n",
       "      <td>B cell</td>\n",
       "      <td>4.116833e-09</td>\n",
       "      <td>0.865333</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.021</td>\n",
       "      <td>1.234877e-03</td>\n",
       "      <td>NC_033676.1</td>\n",
       "      <td>58858363</td>\n",
       "      <td>58860157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Antoine_Bladder_10X</td>\n",
       "      <td>NC-033666.1-87778599-87781699---78-0</td>\n",
       "      <td>NC_033666.1:87778599-87781699</td>\n",
       "      <td>NC_033666.1:87780600-87780897</td>\n",
       "      <td>Microcebus murinus DNA, homologous to human DS...</td>\n",
       "      <td>B cell</td>\n",
       "      <td>5.894225e-07</td>\n",
       "      <td>0.576084</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.029</td>\n",
       "      <td>1.768020e-01</td>\n",
       "      <td>NC_033666.1</td>\n",
       "      <td>87780600</td>\n",
       "      <td>87780897</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample                                    uTAR  \\\n",
       "0  Antoine_Bladder_10X    NC-033671.1-80333049-80333549---18-0   \n",
       "1  Antoine_Bladder_10X    NC-033666.1-51567149-51568449-+-84-0   \n",
       "2  Antoine_Bladder_10X    NC-033672.1-36774049-36774849---67-0   \n",
       "3  Antoine_Bladder_10X  NC-033676.1-58848749-58864449---1010-0   \n",
       "4  Antoine_Bladder_10X    NC-033666.1-87778599-87781699---78-0   \n",
       "\n",
       "                           fasta                     fasta peak  \\\n",
       "0  NC_033671.1:80333049-80333549  NC_033671.1:80333248-80333394   \n",
       "1  NC_033666.1:51567149-51568449  NC_033666.1:51568050-51568213   \n",
       "2  NC_033672.1:36774049-36774849  NC_033672.1:36774301-36774386   \n",
       "3  NC_033676.1:58848749-58864449  NC_033676.1:58858363-58860157   \n",
       "4  NC_033666.1:87778599-87781699  NC_033666.1:87780600-87780897   \n",
       "\n",
       "                                   peak blast result cell type       p value  \\\n",
       "0  Apteryx australis mantelli genome assembly Apt...    B cell  0.000000e+00   \n",
       "1                                       undetermined    B cell  9.602478e-60   \n",
       "2  Microcebus murinus chromosome chr16 clone CH25...    B cell  1.143571e-14   \n",
       "3  PREDICTED: Propithecus coquereli sialophorin (...    B cell  4.116833e-09   \n",
       "4  Microcebus murinus DNA, homologous to human DS...    B cell  5.894225e-07   \n",
       "\n",
       "   avg. logFC  % expression in cell type  % expression in other cells  \\\n",
       "0    0.801622                        0.4                        0.000   \n",
       "1    0.534839                        0.4                        0.002   \n",
       "2    0.535301                        0.4                        0.013   \n",
       "3    0.865333                        0.4                        0.021   \n",
       "4    0.576084                        0.4                        0.029   \n",
       "\n",
       "   p value adj.        chrom     start       end  \n",
       "0  0.000000e+00  NC_033671.1  80333248  80333394  \n",
       "1  2.880340e-54  NC_033666.1  51568050  51568213  \n",
       "2  3.430234e-09  NC_033672.1  36774301  36774386  \n",
       "3  1.234877e-03  NC_033676.1  58858363  58860157  \n",
       "4  1.768020e-01  NC_033666.1  87780600  87780897  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "start_end = utar_df[col_to_use].str.split(':').str[-1].str.split('-', expand=True)\n",
    "utar_df[['start', 'end']] = start_end\n",
    "utar_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['NC-033681.1-16376149-16381299-+-109712-0',\n",
       "       'NC-033678.1-77818149-77821899---250-0',\n",
       "       'NC-033677.1-45626799-45642549-+-1538-0',\n",
       "       'NC-033664.1-8257299-8258649---65581-0',\n",
       "       'NC-033667.1-8459149-8461949---79087-0',\n",
       "       'NC-033683.1-1184799-1186149---9995-0',\n",
       "       'NC-033660.1-41828049-41830249-+-1155-0',\n",
       "       'NC-033662.1-2840399-2840899---166-0',\n",
       "       'NC-033661.1-118276699-118278399-+-160-0',\n",
       "       'NC-033667.1-110239199-110239649-+-44-0'], dtype=object)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "utar_df.uTAR.sample(10).values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     0\n",
       "0  NaN\n",
       "1  NaN\n",
       "2  NaN\n",
       "3  NaN\n",
       "4  NaN"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "utar_df.uTAR.head().str.extract('\\d+-(-\\+)-\\d+-0')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NW_017968608.1\t2815\t2898\tNW-017968608.1-2349-3349---197360-0\t.\t-\n",
      "NC_033669.1\t97217022\t97217242\tNC-033669.1-97215549-97217949---11113-0\t.\t-\n",
      "NC_033666.1\t16147071\t16147137\tNC-033666.1-16147049-16147399-+-8293-0\t.\t+\n",
      "NC_033662.1\t92200223\t92200573\tNC-033662.1-92197399-92201049-+-14496-0\t.\t+\n",
      "NC_033662.1\t2176702\t2176806\tNC-033662.1-2176549-2177049---10702-0\t.\t-\n",
      "NC_033661.1\t123405522\t123405836\tNC-033661.1-123404399-123407549---32935-0\t.\t-\n",
      "NC_033690.1\t45808878\t45809065\tNC-033690.1-45808699-45809599---6693-0\t.\t-\n",
      "NC_033663.1\t17919526\t17919639\tNC-033663.1-17918799-17920049-+-5142-0\t.\t+\n",
      "NC_033684.1\t27037371\t27037777\tNC-033684.1-27034299-27037999---14671-0\t.\t-\n",
      "NC_033666.1\t6768491\t6768652\tNC-033666.1-6767949-6769499-+-442-0\t.\t+\n"
     ]
    }
   ],
   "source": [
    "! head /home/olga/data_sm/tabula-microcebus/analyses/de-novo-orthology/uTAR/cellTypeMarkersUTarsPeaks_deduplicated.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "outdir = '/home/olga/data_sm/tabula-microcebus/analyses/de-novo-orthology/uTAR/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(11408, 6)\n",
      "(3933, 6)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/olga/anaconda/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n",
      "/home/olga/anaconda/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/olga/anaconda/lib/python3.7/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>fasta peak</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4508</th>\n",
       "      <td>NC_033660.1</td>\n",
       "      <td>100376411</td>\n",
       "      <td>100376590</td>\n",
       "      <td>NC_033660.1__100376411-100376590</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4468</th>\n",
       "      <td>NC_033660.1</td>\n",
       "      <td>100391336</td>\n",
       "      <td>100391609</td>\n",
       "      <td>NC_033660.1__100391336-100391609</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>319</th>\n",
       "      <td>NC_033660.1</td>\n",
       "      <td>101998727</td>\n",
       "      <td>101998887</td>\n",
       "      <td>NC_033660.1__101998727-101998887</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11240</th>\n",
       "      <td>NC_033660.1</td>\n",
       "      <td>102218570</td>\n",
       "      <td>102218624</td>\n",
       "      <td>NC_033660.1__102218570-102218624</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3383</th>\n",
       "      <td>NC_033660.1</td>\n",
       "      <td>103462886</td>\n",
       "      <td>103463114</td>\n",
       "      <td>NC_033660.1__103462886-103463114</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             chrom      start        end                        fasta peak  \\\n",
       "4508   NC_033660.1  100376411  100376590  NC_033660.1__100376411-100376590   \n",
       "4468   NC_033660.1  100391336  100391609  NC_033660.1__100391336-100391609   \n",
       "319    NC_033660.1  101998727  101998887  NC_033660.1__101998727-101998887   \n",
       "11240  NC_033660.1  102218570  102218624  NC_033660.1__102218570-102218624   \n",
       "3383   NC_033660.1  103462886  103463114  NC_033660.1__103462886-103463114   \n",
       "\n",
       "      score strand  \n",
       "4508      .      .  \n",
       "4468      .      .  \n",
       "319       .      .  \n",
       "11240     .      .  \n",
       "3383      .      .  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "utar_bed = utar_df[['chrom', 'start', 'end', 'fasta peak']]\n",
    "utar_bed['fasta peak'] = utar_bed['fasta peak'].str.replace(':', \"__\")\n",
    "utar_bed['score'] = '.'\n",
    "utar_bed['strand'] = '.'\n",
    "utar_bed.head()\n",
    "\n",
    "utar_bed = utar_bed.sort_values(['chrom', 'start', 'end'])\n",
    "print(utar_bed.shape)\n",
    "utar_bed = utar_bed.drop_duplicates()\n",
    "print(utar_bed.shape)\n",
    "utar_bed.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "bed = os.path.join(outdir, 'mlca_utars_jan2021.bed')\n",
    "utar_bed.to_csv(bed, index=False, header=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:root] *",
   "language": "python",
   "name": "conda-root-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
